pkg/factory/risingwave_object_factory.go

```diff
@@ -843,6 +843,28 @@ func mergeComponentGroupTemplates(base, overlay *risingwavev1alpha1.RisingWaveCo
 	}
 	r.NodeSelector = mergeMap(r.NodeSelector, overlay.NodeSelector)
 	r.PodTemplate = mergeValue(r.PodTemplate, overlay.PodTemplate)
+	if len(overlay.Tolerations) != 0 {
+		r.Tolerations = overlay.Tolerations
+	}
+	r.PriorityClassName = mergeValue(r.PriorityClassName, overlay.PriorityClassName)
```

Comment:
Not sure if `mergeValue` is the right strategy here. Maybe we should use the higher value?

```
priorityClassName <string>
     If specified, indicates the pod's priority. "system-node-critical" and
     "system-cluster-critical" are two special keywords which indicate the
     highest priorities with the former being the highest priority. Any other
     name must be defined by creating a PriorityClass object with that name. If
     not specified, the pod priority will be default or zero if there is no
     default.
```

Default prio classes are 

```
k get priorityclass
NAME                      VALUE        GLOBAL-DEFAULT   AGE
system-cluster-critical   2000000000   false            6d3h
system-node-critical      2000001000   false            6d3h
```

My gut-feeling is that `someMergeFunc(system-cluster-critical, system-node-critical)` should return `system-node-critical`, because it is the higher value. What do you think? 

Also note that we may add different prio classes in the future. I don't think it is likely, but we may.

Reply:
Actually, `merge` isn't a good name here. What I want to do is to use values from `base` if the corresponding value in `overlay` is a default (zero) one. I have refactored the function to support that better.